### PR TITLE
Added `resample` method in `GenericTimeSeries`

### LINKED ISSUE #5378: Add `resample` method to TimeSeries
### Description

One useful thing when dealing with timeseries is to resample the data. Pandas provide functionality for this, so it would be nice to include it as a method on TimeSeries

### PR DESCRIPTION
**Pull request by devanshshukla99**
### Description

Fixes #5378

### CODE REVIEW

File: sunpy/timeseries/timeseriesbase.py [hunk 1]
```diff
@@ -445,6 +445,33 @@ def concatenate(self, others, same_source=False, **kwargs):
...
+        `~sunpy.timeseries.TimeSeries`
+            A resampled `~sunpy.timeseries.TimeSeries`.
+
+        Notes
+        -----
+        This method is a wrapper around `~pandas.DataFrame.resample`; all additional
+        keyword arguments are passed to this method.
+        """
+
+        object = GenericTimeSeries(self._data.resample(rule, **kwargs).pad(),
```
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:468]: What does the pad do in this context?

Comment (devanshshukla99) [reply sunpy/timeseries/timeseriesbase.py:468]: It'll fill the missing values with `NaN`

**Review by dstansby — CHANGES_REQUESTED** [review]
There are many other operations one might want to do after resampling a `DataFrame` other than `pad()`, see the docs [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.resample.html). I think we need to have quite a bit of discussion over how we are going to implement this and what the API will be. Thanks for opening the PR, and sorry for not indicating this on the original issue. I'll update the original issue with a comment when it has been discussed, but for now I'm going to block this from being merged.

**Comment by nabobalis** [comment]
I guess the ugly thing would be a string keyword to parse for the method?

**Comment by devanshshukla99** [comment]
> There are many other operations one might want to do after resampling a `DataFrame` other than `pad()`, see the docs [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.resample.html). I think we need to have quite a bit of discussion over how we are going to implement this and what the API will be. Thanks for opening the PR, and sorry for not indicating this on the original issue. I'll update the original issue with a comment when it has been discussed, but for now I'm going to block this from being merged.

was debating myself too, but got swayed towards adding basic func first,
anyway do let me know👍

**Comment by dstansby** [comment]
Thanks for working on this! We decided at the development meeting that we will be transitioning to using astropy TimeSeries at some point, so we want to hold off on implementing any new methods on sunpy TimeSeries until we do that. As such I'm going to close this PR; thanks again for opening it.

**Pull request closed without merging**